### ISSUE TITLE
[FlyoutMenu] Various accessibility bugs

### ISSUE THREAD

**Issue by JoelArmendariz** [carbon-design-system/carbon-addons-iot-react#2229]
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug
The `FlyoutMenu` is missing some accessibility features such as tabbing, keydown, and onBlur (to closed the menu) events. 

### To Reproduce

Go to the `FlyoutMenu` and click the button to open the menu. See that you cannot click off of the menu to close it.

In addition, click off of the component to lose focus and press tab to access the button. Press enter and see that esc does not allow you to close the menu or lose focus. 

### Expected behavior
As a user, I should be able to press esc or click off of the component to closed the menu when focused.

### Environment/versions:

- OS: [e.g. MacOS, Windows]
- Browser: [e.g. chrome, safari]
- `carbon-addons-iot-react` version: [e.g. `v2.60.0`]

### Additional context
This component should act like the `Tooltip` component found here: https://next.carbon-addons-iot-react.com/?path=/story/watson-iot-tooltip--default-bottom
